packages/agoric-cli/src/commands/ec.js

```diff
@@ -1,126 +1,304 @@
-import { Command } from 'commander';
-import { makeRpcUtils, storageHelper } from '../lib/rpc.js';
-import { outputExecuteOfferAction } from '../lib/wallet.js';
+/* global globalThis, process, setTimeout */
+import { execFileSync as execFileSyncAmbient } from 'child_process';
+import { Command, CommanderError } from 'commander';
+import { normalizeAddressWithOptions, pollBlocks } from '../lib/chain.js';
+import { getNetworkConfig, makeRpcUtils } from '../lib/rpc.js';
+import { outputExecuteOfferAction, sendAction } from '../lib/wallet.js';
+
+/** @typedef {import('@agoric/smart-wallet/src/offers.js').OfferSpec} OfferSpec */
 
 /**
  *
  * @param {import('anylogger').Logger} _logger
+ * @param {{
+ *   env?: Record<string, string|undefined>,
+ *   fetch?: typeof window.fetch,
+ *   stdout?: Pick<import('stream').Writable, 'write'>,
+ *   execFileSync?: typeof execFileSyncAmbient,
+ *   delay?: (ms: number) => Promise<void>,
+ * }} [io]
  */
-export const makeEconomicCommiteeCommand = async _logger => {
+export const makeEconomicCommiteeCommand = async (_logger, io = {}) => {
+  const {
+    // Allow caller to provide access explicitly, but
+    // default to conventional ambient IO facilities.
+    env = process.env,
+    stdout = process.stdout,
+    fetch = globalThis.fetch,
+    execFileSync = execFileSyncAmbient,
+    delay = ms => new Promise(resolve => setTimeout(resolve, ms)),
+  } = io;
+
   const ec = new Command('ec').description('Economic Committee commands');
 
+  /** @param {string} literalOrName */
+  const normalizeAddress = literalOrName =>
+    normalizeAddressWithOptions(literalOrName, { keyringBackend: 'test' });
+
+  const show = (info, indent) =>
+    stdout.write(`${JSON.stringify(info, null, indent ? 2 : undefined)}\n`);
+
+  const findContinuingIds = (current, agoricNames) => {
+    // XXX should runtime type-check
+    /** @type {{ offerToUsedInvitation: [string, Amount<'set'>][]}} */
+    const { offerToUsedInvitation: entries } = /** @type {any} */ (current);
+
+    assert(Array.isArray(entries));
+
+    const keyOf = (obj, val) => {
+      const found = Object.entries(obj).find(e => e[1] === val);
+      return found && found[0];
+    };
+
+    const found = [];
+    for (const [offerId, { value }] of entries) {
+      /** @type {{ description: string, instance: unknown }[]} */
+      const [{ description, instance }] = value;
+      if (
+        description === 'charter member invitation' ||
+        /Voter\d+/.test(description)
+      ) {
+        const instanceName = keyOf(agoricNames.instance, instance);
+        found.push({ instance, instanceName, description, offerId });
+      }
+    }
+    return found;
+  };
+
+  const abortIfSeen = (instanceName, found) => {
+    const done = found.filter(it => it.instanceName === instanceName);
+    if (done.length > 0) {
+      console.warn(`invitation to ${instanceName} already accepted`, done);
+      throw new CommanderError(1, 'ELREADY', `already accepted`);
```

Comment:
nice. belongs in `lib/wallet` though.

Reply:
I misspelled ALREADY too.

Reply:
I'm moving `findContinuingIds` to `wallet.js`. `wallet.js` doesn't know about `CommandError`, though so I don't think I should move `abortIfSeen`.